vnda/actions/cart/simulation.ts

```diff
@@ -0,0 +1,27 @@
+import { HttpError } from "../../../utils/http.ts";
+import { AppContext } from "../../mod.ts";
+import type { ShippingMethod } from "../../utils/client/types.ts";
+
+export interface Props {
+  skuId: string;
+  quantity: number;
+  zip: string;
+}
+
+const action = async (
+  props: Props,
+  _req: Request,
+  ctx: AppContext,
+): Promise<ShippingMethod[]> => {
+  const { api } = ctx;
+  const {skuId, quantity, zip} = props
+  
+  if (!skuId || !quantity || !zip) {
+    throw new HttpError(400, "Missing cart cookie");
```

Comment:
use `import { badRequest } from "deco/mod.ts";`

then just use

badRequest({message: "missing cart cookie"}) // or something like that